src/formattedcode/output_cyclonedx.py

```diff
@@ -629,10 +666,33 @@ def from_codebase(cls, codebase):
         """
         Return a CycloneDxBom built from a ScanCode ``codebase``.
         """
-        metadata = CycloneDxMetadata.from_headers(codebase.get_headers())
-        packages = codebase.attributes.packages
-        components = list(CycloneDxComponent.from_packages(packages))
-        dependencies = list(CycloneDxDependency.from_packages(packages, components))
+        components = []
+        dependencies = []
+
+        packages_not_found_message = (
+            "The --cyclonedx-xml option will not output any component/dependency data "
+            "as there are no package data in the present scan. To get package data "
+            "please rerun the scan with --package or --system-package CLI options enabled."
+        )
+        codebase.get_or_create_current_header()
+
+        if hasattr(codebase.attributes, 'packages'):
+            packages = codebase.attributes.packages
+            components = list(CycloneDxComponent.from_packages(packages))
+            dependencies = list(CycloneDxDependency.from_packages(packages, components))
+            codebase_headers = codebase.get_headers()
```

Comment:
I think you can delete this line and then de-indent line 693.

Reply:
Done.